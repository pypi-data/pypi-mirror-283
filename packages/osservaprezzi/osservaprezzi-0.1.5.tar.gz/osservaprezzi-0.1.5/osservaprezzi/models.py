"""Models module."""

from __future__ import annotations

from dataclasses import dataclass
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from datetime import datetime


@dataclass(frozen=True)
class GPSCoordinates:
    """GPS coordinates definition."""

    latitude: float
    longitude: float

    def to_json(self) -> dict[str, float]:
        """Return JSON representation of the object."""
        return {
            "lat": self.latitude,
            "lng": self.longitude,
        }


@dataclass(frozen=True)
class Fuel:
    """Fuel definition."""

    id: int
    price: float
    name: str
    fuel_id: int
    is_self: bool
    insert_date: datetime | None
    validity_date: datetime | None


@dataclass(frozen=True)
# pylint: disable-next=too-many-instance-attributes
class Station:
    """Station definition."""

    id: int
    name: str
    description: str
    fuels: list[Fuel]
    location: GPSCoordinates
    insert_date: datetime | None
    address: str | None
    brand: str | None
    email: str | None
    website: str | None
    phone: str | None
    company: str | None
    services: list[Service] | None


@dataclass(frozen=True)
class Marker:
    """Marker definition."""

    type: str
    extension: str
    content: str


@dataclass(frozen=True)
class Brand:
    """Brand definition."""

    id: int
    name: str
    markers: list[Marker]


@dataclass(frozen=True)
class Service:
    """Service definition."""

    id: str | int
    description: str
