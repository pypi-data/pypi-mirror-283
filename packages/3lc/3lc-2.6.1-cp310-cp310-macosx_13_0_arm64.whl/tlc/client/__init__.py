# Pyarmor 8.5.10 (pro), 005329, 3lc, 2024-07-04T17:42:48.735715
from tlcsaas.pyarmor_runtime_005329 import __pyarmor__
__pyarmor__(__name__, __file__, b'PY005329\x00\x03\n\x00o\r\r\n\x80\x00\x01\x00\x08\x00\x00\x00\x04\x00\x00\x00@\x00\x00\x00\xe1\n\x00\x00\x12\t\x06\x00\xf88w;\xcf\xe1\x90,\xf9&\xea\xb9\x84\xc8\xbb\xb4\x00\x00\x00\x00\x00\x00\x00\x00\xe9\\\xd1\xa0\xc1\\\xb2\x88\xba\x9a\xef\x05\xe8\xde\x02{\x87z\xf0T1CA+\x1e6\x8dW\xa4\xf5~\x9e\xf4m\x8aX\xa1\xa7Z\xc2\xc1\xf31\x97?kP\xcf\xc5\x08\x0e\x1bL\xfa\xddB\x11\xbe^\xcb\x98\xfa\x93~\x87\x81D\xf4\xa8\x08\xe54+\xe0^N\x85\xd1\xad\xe1\xb4\xa9\xaaCV\x82\n\x1b7c\xdb\x13\xf3\x1a\x85ZO!)\xe5j\xfc\x88\x8e\xa0M\x1e\xa2;\xcec\xd5\x88g,0\x93\xfe\x1e\xf0\x8d\x08\x84\x85~\xa3\xcb\x1cV*\xe2\xdf\xacK\x1e\x0e\xaeX\x8c\x7fQ}YI\x93\x03U\x03\xbe\x15\xf4\xe5\x17\x15:\x87\xff\xd2f\xfcVRz8n\x10\x1eL\xc5\xd5F2\x12\x04\x01\x01\x06\xc6\xac\xb1\xff\xa8$lT\xbf\x80\x0eZ\x13\xb5\nVa\xc6\xa1s\xb4\xc0o]\xff\xc9\xe4\r\xdaw\x94P\xa9\x99\xa6\xeaB\x1f\xd8\x9e\x17Cuw\x81\xf3b\xdekdg`\xa9\xba\xf1I\xffaJ\xa3\xc0\xb6\xfe$\xe5]\xd9\x14\x82K\xd4\xd3M\xd6/\x11\xb0\x11\xfa\xce\xc6\xf5\xb5\xca\x85\xed\xe1f\xea\x88\xe1\x04\x15f\xcdM\xc4\xce\xee\xa03\x13\xcf)\xcd"\x16>\xdb9I\xdc\xc7\x01\x13`\xa3\x18\xce\x93\x99\x98\x81\x99\x98\x15\xe6\xfa\xe5\xd7\x87\xbd\xa8L\x8c\x97\x02\xffw\xfd\xde\xfa{y\x8f\xb6\x14I \xdb/\x04\xdb\xfe\xd31\x1d\xa3\xcd\xa7~\xb7\xa7\x15B \xdc\xd5\x9f\xb5U\xe2\xc7\xe9\x88b\\-\tXy7D\x8b\xcf\xa11L\xaf\x07\xaa p\xe2\x93t3\x9c\xd0\xc9\rM2\x94\x14\x03\xb5\x16\x1c\x18\x11YBu\r\xabWe\x88\xc5\x01\xf1\x87\x86\x94\xeb~!\x95\xaa;\xd0z0\x1aO\x8e\x99\xe1\xb8\xb8\x99\xf7WC4\xce\xb9$\x06\x978\xd0%/\x06J\xbf\x92,\x13\xd4\x91\xd4\xc0r\x97\xben\xe6nk\xa8\xdb\x900aS\xe8(\x90\\\xa2\xfb\xaa\xa3\xba\xf8C\x0c9\xab\x1c\xefW\x1c\xd0\x17\xb0\xab\x04\xa1\x8a\x17^\x12\'\xbd\xc7]/BE\xda\x19\x8c9U\xd0\x03`\x9e>\x97\xfc\xec\x92OU\xf0\xe9\x0f\x0f\xdf\xf8\xa5pK+a\x8e\x85\x9f9\x15\x02\x10\x1cg\xech\xc74\xcc3\xe21\xfd\x80\xaf\xfe\xdbo\x91\xe7\x16c.\'t;C\xe4s\xe0 %\xb2\x1a\xcd^F;`\x08W\x00\xdb\xdf\xd4\xec-\xcfc\xdd\x1c\x10ih\x99\xd4.\xaeD\x1fL\x12\xa7\xc6\xfe-\xb8<Ao+\x8c\x97i\xfb\xcc\xa5D-\xb7\xf1\t\x1a4\x9c\xce\x80\x8c/x\x98\xa3\x127\xedb\xd9\xc0\xba=\x7f\x04\xd2I\xf7~Y\x98\xc9\xfa\x04}L\x95\xa9\xa2\xff\x13\xd5\xa4\xf97\xaf\x83\x9c\x91\xcb[&\x7f\xd4\x9c\xfe\xeb\x1b\x96T\x0b\xf8\x19#\x81E\x8ez,\x88\x11\xb9I\x01\x81\x052\xc7b\xf3\x98f\xa8\xde\x9b\xc5n\x91z\x13\x1aR\xe1\xfa\xcc\x07#<\xb9vdd\xdfB}\xb4\xe8\xd7\xb6\x1b|\x89{\x9d\xeef\xca\x94pe\x93\xb8\x83"\xb5\x1br;{P\xba\x02\'\x14\xb2\xf9\xe6\x88\x19\xdd\xac\x8bT\xfa\x03\xf2LN\xe4\x83\xa8\xf8\xae\xb3\xc9\x12\x12(KL\xf0\xd9\xe8\x105\\\xf6*\xd4@\xc4Zal\xa0\xb1Q\xbd\xda\x18.\xa7Jz}\x05o\xab/\x8e\xe4.\x8fgb&\xed\x03\xfe\xf4&\t\xaf\x9fr*\xf8\xf2p\xd1\xc1p\x05L\x90\xf1e}\x8e\xe6\xc2\x92\\h\xaaA@g\x86@\xe9\xbe\xa5\xf5\x00~\xef\x01\xeb\x8b\x14`S\xb5\xb1\tn\xbf\xc4\x19\xad\x13\xa1\xe0\xe5\xc7\xddck[}\xc2\xb76$\x80\xbf_\x19\xb2\xde$\x88\xee\x84PC,\xa2Y\xba\xfa\xd9Bv#U4\x0c\xce\xf0e\xc3\xad9~\x89sB8\x9e\xc7\xba1\x1e\x19{\x143_\xf5N1:}\x87\x05x\xfa\xc4"\x8f\xdc\xcb\xf2>\x0b@\xb5Iu\xdf\x80T\x0b\x8f\x1dj\xef\xaf_\xde\xc7\xcb\x9ak\x90\xd8_e\x00\xf77\xd4\xe4e\xcal\x03Q\xc2~.\x12|Tn\x0b/u\x12h!\x83\x90nnS\x96\xef\xde\xe9jF^y\xe4\xc1g\xe0\x93)a\x8ai\xdf\xf5\x04N\xbd#a\x1fHOH\xad\x03y\x17\x9d^\x17\x01t\xb9_m\x06\xc8\xcfoh\x8b\xdc\x12\xc0\xb8a\x9c\x1a\xd3jc\xbf\xae\xf7\xe9\x1dJ\xf7\x0f\xd4<=\n\x06\xac\xa2\x00\x87\x02\xae\xdd\x80\xd7\xc1@\xb7\xb3Sa\xb6\xe1\xc4Z\x16D\x0e"\xba\xb8\x01U\xab\x14\x99\xbbJ\xf9u\x97\xcc\xacA\xc8z\xbd\xbe{R`\xcc\x1dc\xf4Yv\x9a`7s\xddqwK\x96\xd3\xfa\xfc\xa9R\x1f&$I{;w\x95\rs\xf6\x15\xff\x81\xb4\xe2\xed\x95\x7f?\x17\x10Q\xbc\x13\x86"\x9b=O\x82\xf4\xe4\xea{\x0c\x98\xcfzC\xa8\x05\xd9\x8e\xf60@\xee\xa8!\xa7\xa2\x03\xeb\xb0\xd3\xd0\xd2\xf2h\xd2\xb7\x8e\xbec*\xdeP\xe7\x1f\xc46VPt\xaa\x02\xdfl\xe8\xd0}\xd8qk=\xb4P\x1fA|\xd1\x14k\xe0\xf6\xe6\xdb\xdd\xef\x06\x91\xe6"}6\xac\xdf\x96*?\x18\xc6\xd1\xd1}Puc\x98.\x9a\xea\x11d\x9d\xc1\x921\xfe\x05V\xac\xa4\xfd*\xa8\x03\x942\x995\xda\x15Tf\xebt#6\x84y\xfbG\xe9\x00\xb7\xa2B\xbaKO\'\x94z\xcf7\xca\xc7\x04\xb37\xb5\xcd\xa1D9\n\xdb\xf5\x97y\x06\xb1\xec\x04\xc9aNk\x9a\xfdj|\xef\xd1j\x10-\xaeq\xcc\xb2P\xc6M\x08\xf6\x1c\x142d\xba\x19\xb9\xe5\x87HW++\xde\xdc\xef\x87\x16\xce~\xd9U\xd0\xfcR\x07\xfa\x06\x92\x9c\x99\x9aN\xce\xec1\xaf\xda\xa4 Vk\x7f\x0b\x8f\xc8\xb2y\xa1\x87\xb0\x8b\xb8>\n\'\xd4`\x9dt\x0f\xdc\xbac\x1a\xe84\xb1d\xf27\xces\xce8\x8a\xa5\x88_\xad:=\x99\x8e\xf9d\xa9Z7\xd9c\xf4\xcc\xbc\x05f\xe3\xf0\xa7\xd2\xaa\xab\x93\xc7<\x1bl)t\xcb\xbb\x00Q\x8cj\x9b2\x1a\xe7\xf4\x8bP\xbb\xe6\xb7cu\xf2\x1d\x0b\x08\x1f\xfa\xb2l\xdb\xe6b5\xdcW\x96\xae\x98\xb1\x00\xe3\xf9D0\xaf\xce\xd3\x16\x9f\xf5k\xf0\x9e\x8d\xfb8\xa0\x90\x15{\x04G\x91}Hz\x96\x06T\xae\x9cA\xa4\xeaAK\xe8\x98E\x18>I\xec\xc8\xf6_>\xdb\x9d\x96\x0b^\xbe\xec;\x8dC\x98\xda\xe8\xae\xac\xa8\xa2\xd1\x89\x8ch\x08|\x9eH\xc4\xe3j\xde\x7f\xad\xdb\xb3H\xb1\x87E\x02e1\xa8\x15\x98\x9b\x8d\x0f\x91\xe0y\x80\xd6\xfa\xc8\xfd\t3H\x95\xe2\xf2\xa0\x9d\x86\x0e\x9a$\xea\xbe3\n\xc3]\xc7\xdb\xe7\x89\xdb\xb1\xf2\xe2\xf6\x95{M\x8d5\x80\xb6\xc0\xa9~\x99\xa3N\xc9\x83\x11$Y\x97\xf8\'\xcf\xd6\xe3X\xbdCD\xb7FdP\x90y\x88\x0ey\x14\x19\xd0\x0b\xb9ge\x1f\xa6B\xa2\xa4\xb6\xd5\xb7\xdaK\xaa\xcc\x81|b\x9eP\x8er\xa7\xb0d\x89\xff3;\x02D\xd9\xf7\xcb6\xf1\xdc\x13\xf8\x93\xdb\x1f-]\xf1e\xa9WG\x0b\xfc\xf2 a\nif\xcb\xfea\x7f\x85\xbbi>\x93\xf2.\xfb@+awY\x8do\xf8,\x05\x1f\xb41^\xc6\xec\xbf\x86\x95I\xbc\xe6\xa5\xe6b\xae7 \x14a^\t|\xc6^\xd3\xcb\xbe\xeb\x89\xefz\n\xc6\x85\xdb;q\x1f0\xf8\x82\x1ce\xa4\xc4\xe0\xd3\x0c\xaeE\xa6\x83j\xdb\x1f-\xed\x1ch}^\x02\xe5\x84\x9f\xfa\x846\x9aK\xf9\xbd\x07\x9d\x9dg\xb3@T\x03o\xdfN\xc79\x02\xc8\x00\xf8\x9a\x1f=\xbf\xe9\xee^\xc1\xd6h\x1d\xb0<\xfan\xbf\x98\x8a%\xcb\x0bi\x0f\x9f-\xbfiC\x1e\x94\x1e\xe4v\xbd@Wtq\x8a\xad\x1fNg\t\xd4\xdeN\x08D*\x07\xe6\xf3\xd6\x10\xa2\xe1\xdaX\xa9\xe6\x8e4R]\x17\xec\xdcn\xf5L)O\x8d\x95\xfa\xa1\xd9\xc7\x91Q\xfb\xed\xd5\x03B\x96\xb8L\rH\x9c\xf5\x9f\xc6$\xb8\xdb\xed/e,\xbf0r\xc0\xa6\xf3.\xd1/\x98H\xf4\x0e\xe0]\x16mCOrQ 9\r\xa9\xa6\xd8\'5\x99m\x9aB\xe3\xa7v;\xca\xa8\xe5:\x97\xee\xcay\xf7\x7fJ\xb8p\xe5\xa4\x81\xc8%\xf8\xe3\x80\xfe\xb8@\xf5\x85\r\xeev\xcc%\'\xe0F%fE\xfc\x1d\xae\xf3\xed\xb00\xc4Q\xcb\xe5E\xa9\xff\xd4\x82LQ\x7fs\x8a\x80_\xbcS\xec\x8c\x00\x04\x19M9\xb2\x08x|\xdd\xc9\xc19\x01\xf1\xa6\xae\xbe\x94\x00\xd28\x9d\x82\xa6\xa9\xcd\x85$\x8av\xa7\xa8\x15/\x81\xdc\xa0\xf57\xa5\xe8c\x8ff\xfc\xd1\x83\x81:\x0e\x89\xa7\x90\x1ai}\xdc$w\xec0\x83\xe8\xce\xc5x\x80\x0f\xbf\xae[\xf1\xd8\x9e\xc9\xff\xde\x95\xbf q\x9d51\x16\xe2\xd3b\x89\x8aT\xc8\xbd\xb1\xde\x9f\xc2\xfa/\xe09o\x98w\xbb\xa6\xb2\xf8w\xbe\x1e\xb8\x1a\xa6\xf5;Wg\x1bN\xae\x13\xb1\x112\xc8\x98#\xefs>\xbff\x82\x0e\x82\xcf\xc1\x92z\r[\xa4\xea\xdb\x93\xdde\x03\x18\xfe\xe8H\xd1\xcd\x8e6:\x848\xe0PO\x0f~v&u\x84AH\xba\x9df\xc6\x0e\xfa^\x80\xa8\xa2\xbc[[\xf74\x16\x12[\x191r\x99p)\xbd\xcad\xa2\x85\x96\xdc\x8e\x9a\x9d\xe5\xae\xd3\x9d\x13.\xb0\x92w\xc8\xf5\xea\xccp\xc0\xc9\xf3\xbd\x1d\xae~2P-\x8b\xf3H<\x7fKP\xbc\xbe\xd3\n\x1cAfW\xaa\x17\n\x7f\xf8\x16\x8f\xb2W\xb5\xa4\xc3\x9d\xc5+\xba`\x110\xa4\xb3\xf4QP\xa2Lw\xc7\x9c\xbdZ\xc8\x81r\xc71*\xd8e\xa0[\xb6\xa2o\xab!\x0f\x9e\xfd\xec\xb1+l\xc4\x8doB\xba\xe5\t\x93\xb2\xb25\xf3\x16\xe8\xfe\x15\x15tZ\xae\xb2\xff\xd43a\xa3\xa9\x9db\xf1\xdd\x19N\xa4\x84\xa0\xb5\xa5\xfa\xaf\xcb%^w\x1e\x0f/:\xa0\xfatd`\xe0\x0e\x81\xda\xd4n>cN\xad\n\xa4\x89\x7fXQ\xea\xf53\xe9g`\x8c\x18Y\xa3\x03\x81\xb5\xfaN(.\x8a\xe7\xc8*\x97\xe3\xf8\x199p\x1a\x06\xa5\x15\xd5~\x0eiV\xcb$\x8fP\xa7[\x88\xba\x0c \x88\xd2\x91\x03\x7f\x00iXL\x95\n\t\xe4J9\xfcW5\xbal\xef\x8a\xbc\x95\xfc3\xe9X2\x0c\xae\xd0i\xb7\t\xcb-\xf8<\xde\xe7R\xd5\'\x8e\xfeJ\x93d~\xaaQ\x90\x1c\x0ct\x1f :\xdec\xe0\xedYH\xa3\x8e\xf1\x92\x11\xc1\xb7m;\x0eQz\x1b\x07\xd7\x07v4"aAhn\xd6|\x8a\xe5o\x1c0\xdd\x15\x9c\xd24\xd9\x1eE\x17\x1e\x01r\x9bg\xa0\xeb\xa2\x03\xef\xd1m9\x05j}\x18\x03\x98\xcf\x17\xd9o\xd0\x04\x86!\xdf&\xad\xd4\xa9\xc6\xca\x96n\xd4\xb7\xf3B\xea\x16\x81\x0edJ\xa3(\x96\x8a\xa1\xedp\xdd\xc2\x0b\xc4C\xfb\xcd\xe7\x89>\x98\x12\xe2\xf1\x8a\xb4]/\x9dGy\x91\xff:\xb7?\x16\x08\xa9<0\xb7\x90s\x9bV@\x13B\xd28\xbd(\x93\xc6<\x81\xc1\x8f\x7fp\xddB\xe0=0\t\xc0bH\xb32\xb7LS\xa8?\xf4\t>m\x8cJ\xda\xacN\xf6\xe7\x05\x11\xb4,\xad\x98\x85\xf2Z\xa5\x8ea\x8f_\xe5\xa64\xa6\xa6a\xb8\xb5\x0f\xd3\\\xd3\x13\xd4\xab)\x1a=\xbd\xd4h\x96\xe8\xca\xc1\xc9\x94A\xfaa\xcc\x9e\xf7\xa2\x10\xca\x99\xf7r\xc8\xda\xc7\'\x8eq\xcf\xfaJ\x91{\xd4\xd7Vg\x94\xb6\tn\xe05\xa0\x8a\x0c,\xb1jX\x9ey\xd4QF\xa5\x92\x8aQ;J\xb0\xe3\xa5>H\xed\xba\xc4\xae\x8c{\xc3?u\x10\xba\xecf\xe7\xd3N\xb82k\xc2\xfaG.\x0e\x8baM\x16\x82aS\xc7\xda\xa9')
