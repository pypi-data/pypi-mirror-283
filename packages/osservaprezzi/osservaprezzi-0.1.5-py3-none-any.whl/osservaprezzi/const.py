"""Constants module."""

from __future__ import annotations

DEFAULT_TIMEOUT = 10
URL_API_ENDPOINT = "https://carburanti.mise.gov.it/ospzApi"
PATH_API_BRANDS = "registry/alllogos"
PATH_API_FUELS = "registry/fuels"
PATH_API_ZONES = "search/zone"
PATH_API_SERVICE_AREA = "registry/servicearea/%d"
REQUEST_HEADERS = {
    "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:128.0) Gecko/20100101 Firefox/128.0",
    "Origin": "https://carburanti.mise.gov.it",
    "Host": "carburanti.mise.gov.it",
}
