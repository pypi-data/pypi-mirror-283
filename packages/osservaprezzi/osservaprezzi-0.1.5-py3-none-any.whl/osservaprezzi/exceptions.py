"""Osservaprezzi client exception module."""

from __future__ import annotations


class OsservaprezziError(Exception):
    """Osservaprezzi error."""


class ApiError(OsservaprezziError):
    """Api error."""


class ApiTimeoutError(OsservaprezziError):
    """Api timeout error."""
