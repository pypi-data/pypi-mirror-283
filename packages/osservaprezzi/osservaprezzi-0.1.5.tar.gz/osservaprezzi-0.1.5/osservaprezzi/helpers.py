"""Helpers definition."""

from __future__ import annotations

from datetime import datetime
from typing import Any

from .models import Brand, Fuel, GPSCoordinates, Marker, Service, Station


def brand_from_json(json: dict[str, Any]) -> Brand:
    """Create a brand from JSON."""
    return Brand(
        id=json.get("bandieraId", ""),
        name=json.get("bandiera", ""),
        markers=[marker_from_json(marker) for marker in json.get("logoMarkerList", [])],
    )


def fuel_from_json(json: dict[str, Any]) -> Fuel:
    """Create a fuel from JSON."""
    return Fuel(
        id=json.get("id", 0),
        price=json.get("price", 0),
        name=json.get("name", json.get("description", "")),
        fuel_id=json.get("fuelId", 0),
        is_self=json.get("isSelf", False),
        insert_date=datetime.fromisoformat(json["insertDate"])
        if "insertDate" in json
        else None,
        validity_date=datetime.fromisoformat(json["validityDate"])
        if "validityDate" in json
        else None,
    )


def gps_coordinates_from_json(json: dict[str, Any]) -> GPSCoordinates:
    """Create a GPS Coordinates from JSON."""
    return GPSCoordinates(
        latitude=json.get("lat", 0),
        longitude=json.get("lng", 0),
    )


def marker_from_json(json: dict[str, Any]) -> Marker:
    """Create a marker from JSON."""
    return Marker(
        type=json.get("tipoFile", ""),
        extension=json.get("estensione", ""),
        content=json.get("content", ""),
    )


def service_from_json(json: dict[str, Any]) -> Service:
    """Create a service from JSON."""
    return Service(
        id=json.get("id", ""),
        description=json.get("description", ""),
    )


def station_from_json(json: dict[str, Any]) -> Station:
    """Create a station from JSON."""
    return Station(
        id=json.get("id", 0),
        name=json.get("name", ""),
        description=json.get("nomeImpianto", ""),
        fuels=[fuel_from_json(fuel) for fuel in json.get("fuels", [])],
        location=gps_coordinates_from_json(json.get("location", {})),
        insert_date=datetime.fromisoformat(json["insertDate"])
        if "insertDate" in json
        else None,
        address=json.get("address"),
        brand=json.get("brand"),
        email=json.get("email"),
        website=json.get("website"),
        phone=json.get("phoneNumber"),
        company=json.get("company"),
        services=[service_from_json(service) for service in json.get("services", [])],
    )
