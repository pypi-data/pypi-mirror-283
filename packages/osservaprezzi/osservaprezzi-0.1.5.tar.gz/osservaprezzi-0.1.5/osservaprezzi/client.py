"""Osservaprezzi module."""

from __future__ import annotations

from http import HTTPStatus
import logging
from typing import TYPE_CHECKING, Any

from aiohttp import ClientError, ClientSession

from osservaprezzi.helpers import (
    brand_from_json,
    fuel_from_json,
    station_from_json,
)

from .const import (
    DEFAULT_TIMEOUT,
    PATH_API_BRANDS,
    PATH_API_FUELS,
    PATH_API_SERVICE_AREA,
    PATH_API_ZONES,
    REQUEST_HEADERS,
    URL_API_ENDPOINT,
)
from .exceptions import ApiError, ApiTimeoutError

if TYPE_CHECKING:
    from .models import Brand, Fuel, GPSCoordinates, Station

_LOGGER = logging.getLogger(__name__)


class Osservaprezzi:
    """Osservaprezzi client class."""

    def __init__(self, session: ClientSession) -> None:
        self._session = session

    async def get_brands(self) -> list[Brand]:
        """Get brands."""
        json = await self._get(PATH_API_BRANDS)
        return [brand_from_json(data) for data in json.get("loghi", [])]

    async def get_fuels(self) -> list[Fuel]:
        """Get registered fuels."""
        json = await self._get(PATH_API_FUELS)
        return [fuel_from_json(data) for data in json.get("results", [])]

    async def get_stations(
        self,
        coordinates: GPSCoordinates,
        *,
        radius: int = 5,
        fuel_type: str | None = None,
    ) -> list[Station]:
        """Get stations."""
        payload = {"points": [coordinates.to_json()], "radius": radius}
        if fuel_type is not None:
            payload["fuelType"] = fuel_type
        json = await self._post(PATH_API_ZONES, payload)
        return [station_from_json(data) for data in json.get("results", [])]

    async def get_station(self, station_id: int) -> Station | None:
        """Get station."""
        json = await self._get(PATH_API_SERVICE_AREA % station_id)
        return station_from_json(json) if "id" in json else None

    async def _get(self, path: str) -> dict[str, Any]:
        """Perform a GET request."""
        try:
            url = f"{URL_API_ENDPOINT}/{path}"
            async with self._session.get(
                url, timeout=DEFAULT_TIMEOUT, headers=REQUEST_HEADERS
            ) as response:
                if response.status == HTTPStatus.OK:
                    response_data: dict[str, Any] = await response.json()
                    return response_data
        except TimeoutError as ex:
            raise ApiTimeoutError from ex
        except ClientError as ex:
            raise ApiError from ex

        raise ApiError("Unknown error occurred")

    async def _post(self, path: str, payload: dict[str, Any]) -> dict[str, Any]:
        """Perform a POST request."""
        try:
            url = f"{URL_API_ENDPOINT}/{path}"
            async with self._session.post(
                url,
                json=payload,
                timeout=DEFAULT_TIMEOUT,
                headers=REQUEST_HEADERS,
            ) as response:
                if response.status == HTTPStatus.OK:
                    response_data: dict[str, Any] = await response.json()
                    return response_data
        except TimeoutError as ex:
            raise ApiTimeoutError from ex
        except ClientError as ex:
            raise ApiError from ex

        raise ApiError("Unknown error occurred")
