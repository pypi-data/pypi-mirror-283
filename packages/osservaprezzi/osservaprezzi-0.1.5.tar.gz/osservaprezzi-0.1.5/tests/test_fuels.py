from __future__ import annotations

from typing import TYPE_CHECKING

import pytest

if TYPE_CHECKING:
    from syrupy.assertion import SnapshotAssertion

    from osservaprezzi.client import Osservaprezzi


@pytest.mark.usefixtures("client", "mock_client_get_response")
@pytest.mark.parametrize(
    "response",
    [
        {
            "results": [
                {"id": "1-x", "description": "Benzina"},
                {"id": "1-1", "description": "Benzina (Self)"},
                {"id": "1-0", "description": "Benzina (Servito)"},
                {"id": "2-x", "description": "Gasolio"},
                {"id": "2-1", "description": "Gasolio (Self)"},
                {"id": "2-0", "description": "Gasolio (Servito)"},
                {"id": "3-x", "description": "Metano"},
                {"id": "3-1", "description": "Metano (Self)"},
                {"id": "3-0", "description": "Metano (Servito)"},
                {"id": "4-x", "description": "GPL"},
                {"id": "4-1", "description": "GPL (Self)"},
                {"id": "4-0", "description": "GPL (Servito)"},
                {"id": "323-x", "description": "L-GNC"},
                {"id": "323-1", "description": "L-GNC (Self)"},
                {"id": "323-0", "description": "L-GNC (Servito)"},
                {"id": "324-x", "description": "GNL"},
                {"id": "324-1", "description": "GNL (Self)"},
                {"id": "324-0", "description": "GNL (Servito)"},
            ]
        },
        {
            "results": [
                {
                    "id": 75557219,
                    "price": 2.079,
                    "name": "Benzina",
                    "fuelId": 1,
                    "isSelf": False,
                },
                {
                    "id": 75557218,
                    "price": 1.869,
                    "name": "Benzina",
                    "fuelId": 1,
                    "isSelf": True,
                },
                {
                    "id": 75557221,
                    "price": 1.989,
                    "name": "Gasolio",
                    "fuelId": 2,
                    "isSelf": False,
                },
                {
                    "id": 75557220,
                    "price": 1.779,
                    "name": "Gasolio",
                    "fuelId": 2,
                    "isSelf": True,
                },
            ]
        },
        {
            "results": [
                {
                    "id": 75557219,
                    "price": 2.079,
                    "name": "Benzina",
                    "fuelId": 1,
                    "isSelf": False,
                    "insertDate": "2024-07-02T12:00:38Z",
                    "validityDate": "2024-07-02T11:59:00Z",
                },
                {
                    "id": 75557218,
                    "price": 1.869,
                    "name": "Benzina",
                    "fuelId": 1,
                    "isSelf": True,
                    "insertDate": "2024-07-02T12:00:38Z",
                    "validityDate": "2024-07-02T11:59:00Z",
                },
            ]
        },
    ],
)
async def test_fuels(
    client: Osservaprezzi,
    snapshot: SnapshotAssertion,
) -> None:
    assert snapshot(name="fuels") == await client.get_fuels()
