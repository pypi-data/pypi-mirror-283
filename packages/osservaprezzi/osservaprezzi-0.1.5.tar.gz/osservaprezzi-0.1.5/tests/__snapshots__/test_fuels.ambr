# serializer version: 1
# name: test_fuels[response0][fuels]
  list([
    Fuel(id='1-x', price=0, name='Benzina', fuel_id=0, is_self=False, insert_date=None, validity_date=None),
    Fuel(id='1-1', price=0, name='Benzina (Self)', fuel_id=0, is_self=False, insert_date=None, validity_date=None),
    Fuel(id='1-0', price=0, name='Benzina (Servito)', fuel_id=0, is_self=False, insert_date=None, validity_date=None),
    Fuel(id='2-x', price=0, name='Gasolio', fuel_id=0, is_self=False, insert_date=None, validity_date=None),
    Fuel(id='2-1', price=0, name='Gasolio (Self)', fuel_id=0, is_self=False, insert_date=None, validity_date=None),
    Fuel(id='2-0', price=0, name='Gasolio (Servito)', fuel_id=0, is_self=False, insert_date=None, validity_date=None),
    Fuel(id='3-x', price=0, name='Metano', fuel_id=0, is_self=False, insert_date=None, validity_date=None),
    Fuel(id='3-1', price=0, name='Metano (Self)', fuel_id=0, is_self=False, insert_date=None, validity_date=None),
    Fuel(id='3-0', price=0, name='Metano (Servito)', fuel_id=0, is_self=False, insert_date=None, validity_date=None),
    Fuel(id='4-x', price=0, name='GPL', fuel_id=0, is_self=False, insert_date=None, validity_date=None),
    Fuel(id='4-1', price=0, name='GPL (Self)', fuel_id=0, is_self=False, insert_date=None, validity_date=None),
    Fuel(id='4-0', price=0, name='GPL (Servito)', fuel_id=0, is_self=False, insert_date=None, validity_date=None),
    Fuel(id='323-x', price=0, name='L-GNC', fuel_id=0, is_self=False, insert_date=None, validity_date=None),
    Fuel(id='323-1', price=0, name='L-GNC (Self)', fuel_id=0, is_self=False, insert_date=None, validity_date=None),
    Fuel(id='323-0', price=0, name='L-GNC (Servito)', fuel_id=0, is_self=False, insert_date=None, validity_date=None),
    Fuel(id='324-x', price=0, name='GNL', fuel_id=0, is_self=False, insert_date=None, validity_date=None),
    Fuel(id='324-1', price=0, name='GNL (Self)', fuel_id=0, is_self=False, insert_date=None, validity_date=None),
    Fuel(id='324-0', price=0, name='GNL (Servito)', fuel_id=0, is_self=False, insert_date=None, validity_date=None),
  ])
# ---
# name: test_fuels[response1][fuels]
  list([
    Fuel(id=75557219, price=2.079, name='Benzina', fuel_id=1, is_self=False, insert_date=None, validity_date=None),
    Fuel(id=75557218, price=1.869, name='Benzina', fuel_id=1, is_self=True, insert_date=None, validity_date=None),
    Fuel(id=75557221, price=1.989, name='Gasolio', fuel_id=2, is_self=False, insert_date=None, validity_date=None),
    Fuel(id=75557220, price=1.779, name='Gasolio', fuel_id=2, is_self=True, insert_date=None, validity_date=None),
  ])
# ---
# name: test_fuels[response2][fuels]
  list([
    Fuel(id=75557219, price=2.079, name='Benzina', fuel_id=1, is_self=False, insert_date=datetime.datetime(2024, 7, 2, 12, 0, 38, tzinfo=datetime.timezone.utc), validity_date=datetime.datetime(2024, 7, 2, 11, 59, tzinfo=datetime.timezone.utc)),
    Fuel(id=75557218, price=1.869, name='Benzina', fuel_id=1, is_self=True, insert_date=datetime.datetime(2024, 7, 2, 12, 0, 38, tzinfo=datetime.timezone.utc), validity_date=datetime.datetime(2024, 7, 2, 11, 59, tzinfo=datetime.timezone.utc)),
  ])
# ---
